src/python/pants/backend/experimental/python/lockfile_metadata.py

```diff
@@ -17,51 +18,97 @@
 @dataclass
 class LockfileMetadata:
     invalidation_digest: str | None
+    valid_interpreter_constraints: InterpreterConstraints | None
+
+    def is_valid_for(
+        self,
+        expected_invalidation_digest: str | None,
+        user_interpreter_constraints: InterpreterConstraints,
+        interpreter_universe: Iterable[str],
+    ) -> bool:
+        """Returns True if this `LockfileMetadata` represents a lockfile that can be used in the
+        current execution context.
+
+        A lockfile can be used in the current execution context if `expected_invalidation_digest ==
+        invalidation_digest`, and if `user_interpreter_constraints` matches no interpreters
+        specified by `valid_interpreter_constraints`.
```

Comment:
The second part seems off. Do you mean "if the `user_interpreter_constraints` are compatible with the lockfile's interpreter_constraints`"?

Reply:
Looks like I wrote exactly the opposite of what I meant 🤦